src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -528,29 +528,26 @@ private IEnumerable<LibraryDependency> GetLibraryDependenciesForCentralTransitiv
                     CentralPackageVersion centralPackageVersion = targetFrameworkInformation.CentralPackageVersions[node.Item.Key.Name];
                     Dictionary<string, LibraryIncludeFlags> dependenciesIncludeFlags = _includeFlagGraphs[targetGraph];
 
-                    LibraryIncludeFlags suppressParent = LibraryIncludeFlags.None;
+                    LibraryIncludeFlags suppressParent = LibraryIncludeFlags.All;
 
-                    if (centralPackageTransitivePinningEnabled)
+                    // Centrally pinned dependencies are not directly declared but the intersection of all of the PrivateAssets of the parents that pulled it in should apply to it
+                    foreach (GraphNode<RemoteResolveResult> parentNode in EnumerateParentNodes(node))
                     {
-                        // Centrally pinned dependencies are not directly declared but the PrivateAssets from the top-level dependency that pulled it in should apply to it also
-                        foreach (GraphNode<RemoteResolveResult> parentNode in EnumerateParentNodes(node))
-                        {
-                            LibraryDependency parentDependency = rootNode.Item.Data.Dependencies.FirstOrDefault(i => i.Name.Equals(parentNode.Item.Key.Name, StringComparison.OrdinalIgnoreCase));
-
-                            // A transitive dependency that is a few levels deep won't be a top-level dependency so skip it
-                            if (parentDependency == null || parentDependency.ReferenceType != LibraryDependencyReferenceType.Direct)
-                            {
-                                continue;
-                            }
-
-                            suppressParent |= parentDependency.SuppressParent;
-                        }
+                        LibraryDependency parentDependency = rootNode.Item.Data.Dependencies.FirstOrDefault(i => i.Name.Equals(parentNode.Item.Key.Name, StringComparison.OrdinalIgnoreCase));
 
-                        // If all assets are suppressed then the dependency should not be added
-                        if (suppressParent == LibraryIncludeFlags.All)
+                        // A transitive dependency that is a few levels deep won't be a top-level dependency so skip it
+                        if (parentDependency == null || parentDependency.ReferenceType != LibraryDependencyReferenceType.Direct)
                         {
                             continue;
                         }
+
+                        suppressParent &= parentDependency.SuppressParent;
+                    }
+
+                    // If all assets are suppressed then the dependency should not be added
+                    if (suppressParent == LibraryIncludeFlags.All)
```

Comment:
I think this the code is correct, maybe we could improve its readability by reversing the condition and write it instead like this?:
```
if (suppressParent != LibraryIncludeFlags.All)
{
	yield return new LibraryDependency()
	{
		LibraryRange = new LibraryRange(centralPackageVersion.Name, centralPackageVersion.VersionRange, LibraryDependencyTarget.Package),
		ReferenceType = LibraryDependencyReferenceType.Transitive,
		VersionCentrallyManaged = true,
		IncludeType = dependenciesIncludeFlags[centralPackageVersion.Name],
		SuppressParent = suppressParent,
	}; 
}
```

Reply:
If the intersection yields `LibraryIncludeFlags.All` then exiting in this case means that the `LibraryDependency` will be left out of the `centralTransitiveDependencyGroups` section of the `project.assets.json` file and then during `Pack` it will be left out of the `.nuspec`.  This would only happen if all parents of a package set `PrivateAssets="All"` which means they aren't in the `.nuspec` and neither should this transitively pinned package.

Reply:
I thought this was in the foreach loop. I misread it. 

I agree with @marcin-krystianc that flipping the condition might be more readable than using a continue.